static/js/composebox_typeahead.js

```diff
@@ -779,13 +780,64 @@ export function show_flatpickr(element, callback, default_timestamp, options = {
         dateFormat: "Z",
         formatDate: (date) => formatISO(date),
         disableMobile: true,
+        onKeyDown: (selectedDates, dateStr, instance, event) => {
+            const hotkey = get_keydown_hotkey(event);
+
+            if (hotkey.name === "tab") {
+                const elems = [
+                    instance.selectedDateElem,
+                    instance.hourElement,
+                    instance.minuteElement,
+                    instance.amPM,
+                    $(".flatpickr-confirm")[0],
+                ];
+                const i = elems.indexOf(event.target);
+                const target = elems[(i + 1) % elems.length];
```

Comment:
I can add shift + tab support here to select the previous element, if this is a desired feature.

Reply:
Since we are in a loop, I don't think so it is necessary. But we can add it if it is easy to do so.

Reply:
It's easy, but requires some extra logic because javascript handles modulus a bit differently for negative values.